core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -97,11 +101,17 @@ case class ExceptionFailure(
     description: String,
     stackTrace: Array[StackTraceElement],
     fullStackTrace: String,
-    metrics: Option[TaskMetrics])
+    metrics: Option[TaskMetrics],
+    exception: Option[Throwable] = None)
```

Comment:
Will this break pattern matching on this class? Is it possible to add an `unapply` method to preserve matching for the old signature? I'm rusty on the details of backwards compatibility for case classes.